nginx/server.sls

```diff
@@ -34,6 +34,15 @@ nginx_extra_packages:
   - watch_in:
     - service: nginx_service
 
+/etc/ssl/private:
+  file.directory:
+  - mode: 0400
+  - user: nginx
```

Comment:
This is system-wide directory for ssh keys so on Debian, it has `root:ssl-cert` ownership and `0710/drwx--x---` mode which is what you should keep.
AFAIK master process of nginx runs as root (otherwise it wouldn't be able to bind on privileged ports) and as root it loads SSL certificates and doesn't need nginx permissions.